From: Frederic Weisbecker <fweisbec@gmail.com>
Subject: hw-breakpoints: Remove stub unthrottle callback
Date: Sun Feb 28 20:51:15 2010 +0100
Git-commit: 1e259e0a9982078896f3404240096cbea01daca4
Patch-mainline: v2.6.34-rc2
References: FATE#311653, BNC#702123
Signed-off-by: Tony Jones <tonyj@suse.de>

    hw-breakpoints: Remove stub unthrottle callback
    
    We support event unthrottling in breakpoint events. It means
    that if we have more than sysctl_perf_event_sample_rate/HZ,
    perf will throttle, ignoring subsequent events until the next
    tick.
    
    So if ptrace exceeds this max rate, it will omit events, which
    breaks the ptrace determinism that is supposed to report every
    triggered breakpoints. This is likely to happen if we set
    sysctl_perf_event_sample_rate to 1.
    
    This patch removes support for unthrottling in breakpoint
    events to break throttling and restore ptrace determinism.
    
    Signed-off-by: Frederic Weisbecker <fweisbec@gmail.com>
    Cc: 2.6.33.x <stable@kernel.org>
    Cc: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Cc: K.Prasad <prasad@linux.vnet.ibm.com>
    Cc: Paul Mackerras <paulus@samba.org>

---
 arch/x86/kernel/hw_breakpoint.c |    5 -----
 kernel/hw_breakpoint.c          |    1 -
 2 files changed, 6 deletions(-)

--- a/arch/x86/kernel/hw_breakpoint.c
+++ b/arch/x86/kernel/hw_breakpoint.c
@@ -518,8 +518,3 @@ void hw_breakpoint_pmu_read(struct perf_
 {
 	/* TODO */
 }
-
-void hw_breakpoint_pmu_unthrottle(struct perf_event *bp)
-{
-	/* TODO */
-}
--- a/kernel/hw_breakpoint.c
+++ b/kernel/hw_breakpoint.c
@@ -585,5 +585,4 @@ struct pmu perf_ops_bp = {
 	.enable		= arch_install_hw_breakpoint,
 	.disable	= arch_uninstall_hw_breakpoint,
 	.read		= hw_breakpoint_pmu_read,
-	.unthrottle	= hw_breakpoint_pmu_unthrottle
 };
